extras/ping/ping.go

```diff
@@ -487,10 +493,11 @@ func (p *Pinger) recvICMP(recv chan<- *packet) error {
 			if err := p.conn.SetReadDeadline(time.Now().Add(delay)); err != nil {
 				return fmt.Errorf("%w: %s", errCannotSetReadDeadline, err)
 			}
+
 			n, err := p.conn.Read(buf)
 			if err != nil {
 				var netErr *net.OpError
-				if errors.As(err, &netErr) && netErr.Timeout() {
+				if errors.As(err, &netErr) && netErr.Timeout() || err.Error() == "i/o timeout" {
```

Comment:
Please, make sure we are matching the end of the error message by using `strings.HasSuffix`. It's quite common in Go to chain error messages and prefix additional context. For this reason, it would probably be better to try and use `strings.HasSuffix`. I would also recommend defining the specific error string in a constant rather than using the raw string directly inside the check.